File: 0563-binary-tree-tilt/0563-binary-tree-tilt.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = 0

    def findTilt(self, root: Optional[TreeNode]) -> int:
        def calc(root):
            if not root: return 0

            l = calc(root.left)    
            r = calc(root.right)

            self.res += abs(l-r)
            return l + r + root.val

        calc(root)
        return self.res

File: 0563-binary-tree-tilt/0563-binary-tree-tilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res = 0;

    public int calc(TreeNode root) {
        if (root == null) return 0;
        int l = this.calc(root.left);
        int r = this.calc(root.right);

        this.res += Math.abs(l-r);
        return l + r + root.val;
    }

    public int findTilt(TreeNode root) {
        this.calc(root);
        return this.res;
    }
}